File: toyvm/ToyVMDelegate.m
//
//  ToyVMDelegate.m
//  toyvm
//
//  Created by Dan Fry on 21/11/2020.
//

#import "ToyVMDelegate.h"

@implementation ToyVMDelegate

- (id)init {
    self = [super init];
    if (self) {
        self.error = nil;
    }
    return self;
}

- (void)guestDidStopVirtualMachine:(VZVirtualMachine *)virtualMachine {
    self.error = nil;
    CFRunLoopStop(CFRunLoopGetMain());
}

- (void)virtualMachine:(VZVirtualMachine *)virtualMachine didStopWithError:(NSError *)error {
    self.error = error;
    CFRunLoopStop(CFRunLoopGetMain());
}

@end

File: toyvm/ToyVMDelegate.swift
//
//  ToyVMDelegate.swift
//  toyvm
//

import CoreFoundation
import Virtualization

class ToyVMDelegate: NSObject, VZVirtualMachineDelegate {
    var error: Error?

    func guestDidStop(_ virtualMachine: VZVirtualMachine) {
        error = nil
        CFRunLoopStop(CFRunLoopGetMain())
    }

    func virtualMachine(_ virtualMachine: VZVirtualMachine, didStopWithError error: Error) {
        self.error = error
        CFRunLoopStop(CFRunLoopGetMain())
    }
}
